pkg/database/database.go

```diff
@@ -32,18 +35,33 @@ type Database struct {
 	resourceTableName string
 }
 
-func NewDatabase() (*Database, error) {
+func GetDatabaseConnection() (*Database, error) {
 	dataSource, err := ConnectionStr()
 	if err != nil {
 		return nil, err
 	}
-	db, err := otelsql.Open("postgres", dataSource)
-	if err != nil {
-		return nil, err
+	var db *sql.DB
+	configs := []retry.Option{
+		retry.Attempts(10),
+		retry.OnRetry(func(n uint, err error) {
+			log.Printf("Retry request %d, get error: %v", n+1, err)
+		}),
+		retry.Delay(time.Second),
 	}
-	if err = db.Ping(); err != nil {
-		return nil, err
+
+	errRetry := retry.Do(
+		func() error {
+			db, err = otelsql.Open("postgres", dataSource)
+			if err = db.Ping(); err != nil {
+				return err
+			}
+			return err
+		},
```

Comment:
Having both returns the same (`return err`) is confusing, I suggest simplifying it to something like this:

```suggestion
			db, err = otelsql.Open("postgres", dataSource)
			if err != nil {
				return db.Ping()
			}
			return nil
		},
```

Reply:
That  confuses me too. Are you saying, @beerparty , that if the open fails you then want to do a ping?  I would assume you would only ping if the open was successful.

Why ping if the open was successful?

Reply:
The code that you mentioned, it was copied directly from the [go documentation](https://go.dev/doc/database/open-handle#confirm_connection). I've tried an approach as you mentioned but the [RetryableFunc](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#RetryableFunc) that I'm using has to return an error.
But, I think a I can do something with the errors that I'm returning. what do you think?
```
func() error {
    db, err = otelsql.Open("postgres", dataSource)
    if errPing := db.Ping(); errPing != nil {
        return errPing
    }
    return err
}
```

Reply:
Regardless of what the documentation says... shouldn't we error on `otelsql.Open` first, and then try to ping and then error on the ping? It makes sense to me.

Also, from the documentation:

> Your code should check for an error from attempting to create a handle, such as with sql.Open. This won’t be a connection error. Instead, you’ll get an error if sql.Open was unable to initialize the handle. This could happen, for example, if it’s unable to parse the DSN you specified.

Reply:
according to the previous comments maybe something like this it's more clear and should do the same:
```suggestion
			db, err = otelsql.Open("postgres", dataSource)
			if err != nil {
				return err
			}
			return db.Ping()
		},
```

Reply:
It's still not clear to me why we need to call `db.Ping` if the open is successful.  That sort of goes against any pattern I have seen, ever.

Reply:
Never mind, I finally read the referenced documentation and it explains why.

Reply:
@maruiz93 I'm gonna test your suggestion :) Thanks!
@ggallen I understood that the ping is a way to confirm that you can connect to the db when you will not use the connection immediately. 
@rh-hemartin you're right.

Reply:
Changed and tested.